### PR TITLE
add ability to split arrays

### PR DESCRIPTION
**Pull request by talevy**
in response to: https://github.com/elasticsearch/logstash/issues/2131

update to: https://github.com/elasticsearch/logstash/pull/787 which was originally created by @20goto10

### CODE REVIEW

**Comment by suyograo** [comment]
@talevy does this create a new `Event` for every element in the array?

File: lib/logstash/filters/split.rb [hunk 1]
```diff
@@ -48,7 +56,7 @@ def filter(event)
 
       event_split = event.clone
```
Comment (talevy) [inline lib/logstash/filters/split.rb]: @suyograo event is cloned here to be split into an independent event

File: spec/filters/split_spec.rb [hunk 2]
```diff
@@ -56,4 +57,49 @@
...
+
+    sample("array" => ["big", "bird", "sesame street"]) do
+      insist { subject.length } == 3
+      insist { subject[0]["element"] } == "big"
+      insist { subject[1]["element"] } == "bird"
+      insist { subject[2]["element"] } == "sesame street"
+    end
+  end
+
+  describe "Split Filter" do
```
Comment (jsvd) [inline spec/filters/split_spec.rb]: this describe block is similar to its parent "describe LogStash::Filters::Split"

maybe 
```ruby
context "when invalid type is passed" do
  it "should raise exception" do
   # ...
  end
end
```

File: lib/logstash/filters/split.rb [hunk 3]
```diff
@@ -33,11 +37,15 @@ def filter(event)
     original_value = event[@field]
 
     # If for some reason the field is an array of values, take the first only.
```
Comment (jsvd) [inline lib/logstash/filters/split.rb]: update this comment to match the code

File: spec/filters/split_spec.rb (hunk 2 above)
Comment (talevy) [inline spec/filters/split_spec.rb]: you're right, just tried to keep it consistent with the previously existing tests

File: lib/logstash/filters/split.rb (hunk 3 above)
Comment (talevy) [inline lib/logstash/filters/split.rb]: updated/removed irrelevent comment

File: lib/logstash/filters/split.rb [hunk 4]
```diff
@@ -32,12 +36,15 @@ def filter(event)
...
-
-    # Using -1 for 'limit' on String#split makes ruby not drop trailing empty
-    # splits.
-    splits = original_value.split(@terminator, -1)
+    if original_value.is_a?(Array)
+      splits = original_value
+    elsif original_value.is_a?(String)
+      # Using -1 for 'limit' on String#split makes ruby not drop trailing empty
+      # splits.
+      splits = original_value.split(@terminator, -1)
```
Comment (suyograo) [inline lib/logstash/filters/split.rb]: can we call this variable `@delimeter`..

Comment (suyograo) [inline lib/logstash/filters/split.rb]: nevermind, its a config variable. needs to be backward compatible

**Comment by suyograo** [comment]
LGTM

**Pull request merged**